### ISSUE TITLE
Wierd behavior with inproc socket

### ISSUE THREAD

**Issue by ppatelclearpool** [zeromq/jeromq#225]
Version=0.3.4

**Issue**
In my thread below, I call the send method and the thread never regains control after that method.  It's hung there forever.  One interesting point is that the thread state is RUNNABLE and i noticed my CPU pegged at 100%.

```java
"Thread-5" prio=10 tid=0x00007fa99c803800 nid=0x29aa runnable [0x00007fa90c46d000]
   java.lang.Thread.State: RUNNABLE
        at sun.nio.ch.EPollArrayWrapper.epollWait(Native Method)
        at sun.nio.ch.EPollArrayWrapper.poll(EPollArrayWrapper.java:269)
        at sun.nio.ch.EPollSelectorImpl.doSelect(EPollSelectorImpl.java:79)
        at sun.nio.ch.SelectorImpl.lockAndDoSelect(SelectorImpl.java:87)
        - locked <0x000000069d072cf8> (a sun.nio.ch.Util$2)
        - locked <0x000000069d072ce0> (a java.util.Collections$UnmodifiableSet)
        - locked <0x000000069d17ec90> (a sun.nio.ch.EPollSelectorImpl)
        at sun.nio.ch.SelectorImpl.select(SelectorImpl.java:98)
        at zmq.Signaler.wait_event(Signaler.java:135)
        at zmq.Mailbox.recv(Mailbox.java:105)
        at zmq.SocketBase.process_commands(SocketBase.java:793)
        at zmq.SocketBase.send(SocketBase.java:635)
        at org.zeromq.ZMQ$Socket.send(ZMQ.java:1206)
        at org.zeromq.ZFrame.send(ZFrame.java:141)
        at org.zeromq.ZFrame.sendAndKeep(ZFrame.java:156)
        at org.zeromq.ZMsg.send(ZMsg.java:186)
        at org.zeromq.ZMsg.send(ZMsg.java:161)
```

**Comment by daveyarwood** [comment]
[Looks like we're compiling with 1.7 now](https://github.com/zeromq/jeromq/blob/master/pom.xml#L74-L75).

@ppatelclearpool Is your issue resolved?

**Comment by ppatelclearpool** [comment]
@daveyarwood  - the issue i posted above was solved, but we faced another bug when using inproc zmq socket.  Thread1 tried to communicate thread2 which had a selector but was not returning any elements (i think theres a ticket open for this https://github.com/zeromq/jeromq/issues/283).  We attempted to fix it, but had no luck.  I love the idea of JeroMq, but it's too buggy to use for us at this point.
